vllm/worker/worker.py

```diff
@@ -322,10 +322,10 @@ def prepare_worker_input(
         # `blocks_to_swap_in` and `blocks_to_swap_out` are cpu tensors.
         # they contain parameters to launch cudamemcpyasync.
         blocks_to_swap_in = torch.tensor(execute_model_req.blocks_to_swap_in,
-                                         device="cpu",
+                                         device="cuda",
```

Comment:
This won't be compatible with other devices I think?

Reply:
As suggested by Kaichao, we will use a static pinned CPU memory buffer to host this array.